pkg/skaffold/deploy/docker/deploy.go

```diff
@@ -67,18 +74,24 @@ func NewDeployer(ctx context.Context, cfg dockerutil.Config, labeller *label.Def
 		return nil, err
 	}
 
+	debugHelpersRegistry, err := config.GetDebugHelpersRegistry(cfg.GlobalConfig())
```

Comment:
following the pattern of the other deployers: `cfg.GlobalConfig()` is stored as a struct-local variable to be used later on, including in this call to `GetDebugHelpersRegistry()`. the main difference is that in the other deployers, it's called after instantiation, but we call it during instantiation for the Docker deployer because we need to set up the `DebugManager` with it.

the other struct-local variables are being populated using direct calls to the accessors on the `cfg` object, like the `insecureRegistries`, so I'm not sure how this one is different? I'd instantiate it inline with the call to `NewDebugManager()`, except we need to check the error here first.

Reply:
We can do this later, but IMHO this should be `config.GetDebugHelpersRegistry()` with no argument, and the *implementation* should be looking into the global config.  This deployer code should not need to know *how* the registry is being looked up.